deploy/cloudformation/manifest-pipeline-pipeline.yml

```diff
@@ -0,0 +1,769 @@
+            Resource: !Sub 'arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:*'
+            Effect: Allow
+          - Action:
+              - 'apigateway:*'
+            Resource: !Sub 'arn:aws:apigateway:${AWS::Region}::*'
+            Effect: Allow
+          - Action:
+              - 'iam:GetRole'
+              - 'iam:CreateRole'
+              - 'iam:DeleteRole'
+            Resource: !Sub 'arn:aws:iam::${AWS::AccountId}:role/${AppName}-*'
+            Effect: Allow
+          - Action:
+              - 'iam:AttachRolePolicy'
+              - 'iam:DetachRolePolicy'
+              - 'iam:DeleteRolePolicy'
+              - 'iam:PutRolePolicy'
+            Resource: !Sub 'arn:aws:iam::${AWS::AccountId}:role/${AppName}-*'
+            Effect: Allow
+          - Action:
+              - 'iam:PassRole'
+            Resource:
+              - '*'
+            Effect: Allow
+          - Action:
+              - 'cloudformation:CreateChangeSet'
+            Resource: !Sub >-
+              arn:aws:cloudformation:${AWS::Region}:aws:transform/Serverless-2016-10-31
+            Effect: Allow
+      Roles:
+        - !Ref CloudFormationTrustRole
+  CodePipelineTrustRole:
+    Description: Creating service role in IAM for AWS CodePipeline
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - codepipeline.amazonaws.com
+            Action: 'sts:AssumeRole'
+      Path: /
+  CodePipelineRolePolicy:
+    Type: 'AWS::IAM::Policy'
+    DependsOn: CodePipelineTrustRole
+    Description: Setting IAM policy for the service role for AWS CodePipeline
+    Properties:
+      PolicyName: CodePipelineRolePolicy
+      PolicyDocument:
+        Statement:
+          - Action:
+              - 's3:CreateBucket'
+              - 's3:DeleteBucket'
+              - 's3:PutBucketLogging'
+            Resource:
+              - 'arn:aws:s3:::mellon-manifest-pipeline*'
+            Effect: 'Allow'
+          - Action:
+              - 's3:GetObject'
+              - 's3:GetObjectVersion'
+              - 's3:GetBucketVersioning'
+            Resource:
+              - 'arn:aws:s3:::mellon-manifest-pipeline*/*'
+              - 'arn:aws:s3:::codepipeline*'
+            Effect: 'Allow'
+          - Action:
+              - 's3:PutObject'
+            Resource:
+              - 'arn:aws:s3:::codepipeline*'
+            Effect: Allow
+          - Action:
+              - 'codebuild:StartBuild'
+              - 'codebuild:BatchGetBuilds'
+            Resource: '*'
+            Effect: Allow
+          - Action:
+              - 'cloudwatch:*'
+              - 's3:*'
+              - 'sns:*'
+              - 'cloudformation:*'
+              - 'rds:*'
+              - 'sqs:*'
+              - 'iam:PassRole'
+            Resource: '*'
+            Effect: Allow
+          - Action:
+              - 'lambda:InvokeFunction'
```

Comment:
The pipeline shouldn't need to invoke all functions - is there a specific Lambda that is being leveraged by the CodePipeline? If it's created by this stack, you can just do a `!GetAtt` for narrowing the IAM scope.

Reply:
If I remove InvokeFunction because I can't think of any reason that it is invoking a function but leave list function is that ok?  
I am actually not sure how it would be listing either.